### ISSUE TITLE
Publishing to two repositories with different platform configurations

### ISSUE THREAD

**Issue by JakeWharton** [vanniktech/gradle-maven-publish-plugin#754]
I want to publish to a repo-local maven repo (`build/test-maven/`) for testing, but also publish to Maven Central for releases. This works fine today, except that my test repo requires source and javadoc jars be generated. This means that Dokka has to run in order to test, and we have like 40+ modules so Dokka takes fooooorreeeeevveeeer.

As far as I can tell, the platform mechanism means that I can only choose to include sources/javadoc jars at the project-level once, and that configuration will be used by all publications. I believe that means there's no way to do what I want today then, right?

And if that's the case, maybe we could figure out an evolution to make this possible. Thanks!

**Comment by gabrielittner** [comment]
Making it dynamic based on which repository a publication is published to unfortunately won't work, since publications and repositories are independent from each other and it's not possible to modify a publication just for one repository.

When using the base plugin (or the main plugin but still calling `configure(Platform)`) like you mentioned you can make that decision for a project and you could make it conditional based on a property. What we could do to make this nicer for the main plugin is to have a default boolean Gradle properties for each for sources and javadoc. For programmatic configuration we can also boolean parameters to `configureBasedOnAppliedPlugins` (this is usually called by the main plugin in afterEvaluate and automatically configures the right platform if none was manually set).

**Comment by JakeWharton** [comment]
Yeah I don't think it needs to be dynamic. I'm okay with two publications targeted at the two repositories but sourced basically from the same tasks. One would only build the main artifacts, and the other would build the main artifacts and doc jars and source jars.

I guess the goal is to not have to duplicate _too_ much where things like the coordinates and pom config would still be shared.

**Comment by gabrielittner** [comment]
As a simple first step using the same publications and everything, just having a config, you can do the following in the current release:
```
subprojects {
    plugins.withId("com.vanniktech.maven.publish") {
        extensions.configure(MavenPublishBaseExtension::class.java) {
            configureBasedOnAppliedPlugins(sourcesJar = false, javadocJar = false)
        }
    }
}
```

Caveats:
- `configureBasedOnAppliedPlugins` needs to be called after other plugins that determine what gets published are applied (e.g. Kotlin or Android plugins) but before `afterEvaluate`. So for the snippet above the publish plugin would need be applied after the Java/Kotlin/Android plugin.
- For some project types it might still produce empty source jars, but since the goal is to save time that should be fine.
- Doesn't have an effect on projects using `com.gradle.plugin-publish`

Would be interesting to know whether this is already helpful.
